source/class/qx/tool/compiler/cli/commands/package/Publish.js

```diff
@@ -26,102 +26,111 @@ const glob = require("glob");
-            alias: "q",
-            describe: "No output"
-          },
-
-          message: {
-            alias: "m",
-            type: "string",
-            describe: "Set commit/release message"
-          },
-
-          dryrun: {
-            type: "boolean",
-            describe: "Deprecated. Use --dry-run"
-          },
-
-          "dry-run": {
-            type: "boolean",
-            alias: "d",
-            describe: "Show result only, do not publish to GitHub"
-          },
-
-          force: {
-            type: "boolean",
-            alias: "f",
-            describe: "Ignore warnings (such as demo check)"
-          },
-
-          "create-index": {
-            type: "boolean",
-            alias: "i",
-            describe:
-              "Create an index file (qooxdoo.json) with paths to Manifest.json files"
-          },
-
-          "qx-version": {
-            type: "string",
-            check: argv => semver.valid(argv.qxVersion),
-            describe:
-              "A semver string. If given, the qooxdoo version for which to publish the package"
-          },
-
-          breaking: {
-            type: "boolean",
-            describe:
-              "Do not create a backwards-compatible release, i.e. allow compatibility with current version only"
-          },
-
-          "qx-version-range": {
-            type: "string",
-            describe:
-              "A semver range. If given, it overrides --qx-version and --breaking and sets this specific version range"
-          }
-        }
-      };
+    async createCliCommand(clazz = this) {
+      let cmd = await qx.tool.compiler.cli.Command.createCliCommand(clazz);
+      cmd.set({
+        name: "publish",
+        description: "publishes a new release of the package on GitHub. Requires a GitHub access token. By default, makes a patch release."
+      });
+
+      cmd.addFlag(
+        new qx.tool.cli.Flag("type").set({
+          shortCode: "t",
+          description: "Set the release type",
+          type: ["major", "premajor", "minor", "preminor", "patch", "prepatch", "prerelease"]
+        })
+      );
+
+      cmd.addFlag(
+        new qx.tool.cli.Flag("noninteractive").set({
+          shortCode: "I",
+          description: "Do not prompt user",
+          type: "boolean"
+        })
+      );
+
+      cmd.addFlag(
+        new qx.tool.cli.Flag("use-version").set({
+          shortCode: "V",
+          description: "Use given version number",
+          type: "string"
+        })
+      );
+
+      cmd.addFlag(
+        new qx.tool.cli.Flag("prerelease").set({
+          shortCode: "p",
+          description: "Publish as a prerelease (as opposed to a stable release)",
+          type: "boolean"
+        })
+      );
+
+      cmd.addFlag(
+        new qx.tool.cli.Flag("message").set({
+          shortCode: "m",
+          description: "Set commit/release message",
+          type: "string"
+        })
+      );
+
+      cmd.addFlag(
```

Comment:
Should we remove deprecated flag in 8.0 v?

Reply:
yes we should. 
Done